### PR TITLE
Regular OT upmerge to `3873c6f`

### PR DESCRIPTION
**Pull request by canisLupus1313**
Openthread upmerge plus implementation of new OT options and missing API.

### CODE REVIEW

File: modules/openthread/platform/ble.c [hunk 1]
```diff
@@ -388,6 +395,47 @@ static void bt_ready(int err)
...
+
+	aBleLinkCapabilities->mGattNotifications = 1;
+	aBleLinkCapabilities->mL2CapDirect = 0;
+	aBleLinkCapabilities->mRsv = 0;
+
+}
+
+bool otPlatBleSupportsMultiRadio(otInstance *aInstance)
+{
+    OT_UNUSED_VARIABLE(aInstance);
```
Comment (pdgendt) [inline modules/openthread/platform/ble.c:410]: Nit: alignment

Comment (canisLupus1313) [reply modules/openthread/platform/ble.c]: fixed.

**Review by rlubos — DISMISSED** [review]

**Review by edmont — DISMISSED** [review]

**Review by maciejbaczmanski — DISMISSED** [review]

File: modules/openthread/CMakeLists.txt [hunk 2]
```diff
@@ -44,11 +44,13 @@ kconfig_to_ot_option(CONFIG_OPENTHREAD_BACKBONE_ROUTER_DUA_NDPROXYING OT_BACKBON
 kconfig_to_ot_option(CONFIG_OPENTHREAD_BACKBONE_ROUTER_MULTICAST_ROUTING OT_BACKBONE_ROUTER_MULTICAST_ROUTING "Enable BBR MR support")
 kconfig_to_ot_option(CONFIG_OPENTHREAD_BLE_TCAT OT_BLE_TCAT "Enable BLE TCAT support")
 kconfig_to_ot_option(CONFIG_OPENTHREAD_BORDER_AGENT OT_BORDER_AGENT "Enable Border Agent")
+kconfig_to_ot_option(CONFIG_OPENTHREAD_BORDER_AGENT_EPHEMERAL_KEY_ENABLE OT_BORDER_AGENT_EPSKC "Border agent ephemeral PSKc")
 kconfig_to_ot_option(CONFIG_OPENTHREAD_BORDER_ROUTER OT_BORDER_ROUTER "Enable Border Router")
 kconfig_to_ot_option(CONFIG_OPENTHREAD_BORDER_ROUTING OT_BORDER_ROUTING "Enable Border routing")
 kconfig_to_ot_option(CONFIG_OPENTHREAD_BORDER_ROUTING_COUNTERS OT_BORDER_ROUTING_COUNTERS "Enable Border routing counters")
 kconfig_to_ot_option(CONFIG_OPENTHREAD_BORDER_ROUTING_DHCP6_PD OT_BORDER_ROUTING_DHCP6_PD "DHCPv6-PD support in border routing")
 kconfig_to_ot_option(CONFIG_OPENTHREAD_CHANNEL_MANAGER OT_CHANNEL_MANAGER "Enable channel manager support")
+kconfig_to_ot_option(CONFIG_OPENTHREAD_CHANNEL_MANAGER_CSL OT_CHANNEL_MANAGER_CSL "Channel manager for csl channel")
```
[the remark this answers is not in the document]
Comment (canisLupus1313) [reply modules/openthread/CMakeLists.txt]: fixed.

File: modules/openthread/Kconfig.features [hunk 3]
```diff
@@ -65,6 +68,9 @@ config OPENTHREAD_CHANNEL_MANAGER
 	bool "Channel manager support"
 	depends on OPENTHREAD_CHANNEL_MONITOR
 
+config OPENTHREAD_CHANNEL_MANAGER_CSL
+	bool "Channel manager for csl channel"
```
Comment (edmont) [inline modules/openthread/Kconfig.features:72]:
```suggestion
	bool "Channel manager for CSL channel"
```

**Review by edmont — COMMENTED** [review]
Somehow my previous suggestion here was not added to the review.

File: modules/openthread/Kconfig.features (hunk 3 above)
Comment (canisLupus1313) [reply modules/openthread/Kconfig.features]: fixed.

**Review by maciejbaczmanski — DISMISSED** [review]

File: west.yml [hunk 4]
```diff
@@ -306,7 +306,7 @@ manifest:
       revision: da78aea63159771956fe0c9263f2e6985b66e9d5
       path: modules/lib/open-amp
     - name: openthread
-      revision: 49c59ec519cc8b49dd58978d1bc80b7ae7ba88d0
+      revision: pull/158/head
```
Comment (rlubos) [inline west.yml:309]: Please update  SHA `3873c6fcd5a8a9dd01b71e8efe32ef5dc7923bb1`

Comment (canisLupus1313) [reply west.yml]: Done.

**Review by rlubos — APPROVED** [review]

**Review by pdgendt — APPROVED** [review]

**Pull request merged**